### PR TITLE
support for AIX and Solaris in cron_allow resource

### PR DESCRIPTION
**Pull request by aklyachkin**
Signed-off-by: aklyachkin <aklyachkin@gmail.com>

Issue #9785 

## Description
Support for AIX and Solaris in cron_allow resource

## Related Issue
#9785 

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [X] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.
- [X] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: lib/chef/resource/cron_access.rb [hunk 1]
```diff
@@ -76,9 +81,14 @@ class CronAccess < Chef::Resource
 
       action :deny do
         description "Add the user to the cron.deny file."
+        deny_path = value_for_platform_family(
+            'aix' => '/var/adm/cron/cron.deny',
+            'solaris' => '/etc/cron.d/cron.deny',
+            'default' => '/etc/cron.deny'
```
Comment (lamont-granquist) [inline lib/chef/resource/cron_access.rb:87]: this hash is really crying out for being made a constant and the duplication removed.

Comment (aklyachkin) [reply lib/chef/resource/cron_access.rb:87]: which duplication do you mean?

Comment (phiggins) [reply lib/chef/resource/cron_access.rb:87]: The arch to path mapping is repeated. That could be extracted into something like this:

```
CRON_PATHS = {           
  'aix' => '/var/adm/cron/',
  'solaris' => '/etc/cron.d/',
  'default' => '/etc/'
}

# later on ...

allow_path = File.join(value_for_platform_family(CRON_PATHS), "cron.allow")

# even later ...

deny_path = File.join(value_for_platform_family(CRON_PATHS), "cron.deny")
```

Comment (aklyachkin) [reply lib/chef/resource/cron_access.rb]: Thank you @phiggins for the explanation! I updated the file.

File: lib/chef/resource/cron_access.rb [hunk 2]
```diff
@@ -76,9 +83,10 @@ class CronAccess < Chef::Resource
 
       action :deny do
         description "Add the user to the cron.deny file."
+        deny_path = File.join(value_for_platform_family(CRON_PATHS), "cron.deny")
```
Comment (tas50) [inline lib/chef/resource/cron_access.rb:86]: This is causing Test Kitchen failures here https://buildkite.com/chef-oss/chef-chef-master-verify/builds/4249#dc7f75c5-4bfe-4e17-8e2b-ea3ebce0ffd7/953-2471

I think you might want to change both of these to 
```suggestion
        deny_path = ::File.join(value_for_platform_family(CRON_PATHS), "cron.deny")
```

Also can you rebase this branch against master. There's a conflict on the description field now.

**Comment by tas50** [comment]
I pulled this one locally, rebased it, and merged it to master. Thanks @aklyachkin

**Pull request closed without merging**